modin/utils.py

```diff
@@ -561,3 +572,103 @@ def import_optional_dependency(name, message):
             f"Missing optional dependency '{name}'. {message} "
             f"Use pip or conda to install {name}."
         ) from None
+
+
+def _get_modin_deps_info() -> dict[str, JSONSerializable]:
+    """
+    Return Modin-specific dependencies information as a JSON serializable dictionary.
+
+    Returns
+    -------
+    dict[str, JSONSerializable]
+        The dictionary of Modin dependencies and their versions.
+    """
+    import modin  # delayed import so modin.__init__ is fully initialized
+
+    result = {"modin": modin.__version__}
+
+    for pkg_name, pkg_version in [
+        ("ray", MIN_RAY_VERSION),
+        ("dask", MIN_DASK_VERSION),
+        ("distributed", MIN_DASK_VERSION),
+    ]:
+        try:
+            pkg = importlib.import_module(pkg_name)
+        except ImportError:
+            result[pkg_name] = None
+        else:
+            result[pkg_name] = pkg.__version__ + (
+                f" (outdated; >={pkg_version} required)"
+                if version.parse(pkg.__version__) < pkg_version
+                else ""
+            )
+
+    try:
+        # We import ``PyDbEngine`` from this module since correct import of ``PyDbEngine`` itself
+        # from Omnisci is located in it with all the necessary options for dlopen.
+        from modin.experimental.core.execution.native.implementations.omnisci_on_native.utils import (  # noqa
+            PyDbEngine,
+        )
+
+        result["omniscidbe"] = "present"
```

Comment:
Is there a way to know omniscidbe version?

Reply:
I didn't find any (except querying by something like `conda list`). Maybe there's some C API of sorts... @modin-project/modin-omnisci ideas?

Reply:
Currently, we don't have any API to get omniscidbe version

Reply:
Does  `conda list` way make sense to use?

Reply:
Too many assumptions (and `conda list` is a slow thing to me).
I would rather we add an API function to get the version.

Reply:
Also we assume that `conda` is present in the PATH which is not guaranteed.